File: C++/프로그래머스/1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
#include <unordered_set>
#include <algorithm>
using namespace std;

int solution(vector<int> nums) {
    unordered_set<int> unique_pokemon(nums.begin(), nums.end());
    int max_pokemon_count = nums.size() / 2;
    int unique_pokemon_count = unique_pokemon.size();
    
    return min(max_pokemon_count, unique_pokemon_count);
}

File: Python3/프로그래머스/1/1845. 폰켓몬/폰켓몬.py
def solution(nums):
    # 전체 폰켓몬 수
    total_pokemon = len(nums)
    # 선택할 수 있는 최대 폰켓몬 수
    max_selectable = total_pokemon // 2
    # 폰켓몬 종류의 수
    unique_pokemon_types = len(set(nums))
    
    # 최대 선택할 수 있는 종류의 수는 max_selectable과 unique_pokemon_types 중 작은 값
    return min(max_selectable, unique_pokemon_types)
